### PR TITLE
Set additional flags for a3 and a3plus

### PR DESCRIPTION
**Pull request by michelle-yooh**
- add example usage commands for scripts running on a3

### CODE REVIEW

File: end_to_end/gpu/a3/test_llama2_7b.sh [hunk 1]
```diff
@@ -1,5 +1,9 @@
 #!/bin/bash
 
+# Example command to invoke this script
+# export COMMAND="bash end_to_end/gpu/a3/test_llama2_7b.sh"
+# bash gpu_multi_process_run.sh
```
Comment (gobbleturk) [inline end_to_end/gpu/a3/test_llama2_7b.sh:5]: it looks like you have to set more arguments for gpu_multi_process_run - e.g. NNODES, JAX_COORDINATOR_ADDRESS?

**Review by gobbleturk — APPROVED** [review]

File: end_to_end/gpu/a3/test_llama2_7b.sh (hunk 1 above)
Comment (michelle-yooh) [reply end_to_end/gpu/a3/test_llama2_7b.sh:5]: I added these example commands thinking that these scripts can run on outside of GKE environment. It IS feasible but I think it's too complicated for users to figure out the correct argument. I will remove these comments.

File: gpu_multi_process_run.sh [hunk 2]
```diff
@@ -16,55 +16,56 @@ export JAX_COORDINATOR_PORT=$JAX_COORDINATOR_PORT
...
+  if [[ "$USE_GPUDIRECT" == "tcpx" ]] || [[ "$USE_GPUDIRECT" == "fastrak" ]]; then
+    export CUDA_DEVICE_MAX_CONNECTIONS=1
+    export NCCL_ALGO=Ring,Tree
     export NCCL_CROSS_NIC=0
-    export NCCL_ALGO=Ring
-    export NCCL_PROTO=Simple
     export NCCL_DEBUG=INFO
-    export NCCL_NET_GDR_LEVEL=PIX
-    export NCCL_P2P_PXN_LEVEL=0
     export NCCL_DEBUG_SUBSYS=INIT,GRAPH,ENV,TUNING,NET,VERSION
```
Comment (yangyuwei) [inline gpu_multi_process_run.sh:24]: Have you verified this for A3+? I only used `NCCL_DEBUG_SUBSYS=INIT,NET` and `NCCL_DEBUG_SUBSYS=ALL` in my experiments on A3+.

I think we can remove this flag for now and add it in ad-hoc runs if we need more NCCL logs for debugging.

File: gpu_multi_process_run.sh [hunk 3]
```diff
@@ -16,55 +16,56 @@ export JAX_COORDINATOR_PORT=$JAX_COORDINATOR_PORT
 export JAX_COORDINATOR_ADDRESS=$JAX_COORDINATOR_ADDRESS
 
 set_nccl_gpudirect_tcpx_specific_configuration() {
-  if [[ "$USE_GPUDIRECT" == "tcpx" ]]; then
-    echo "Using GPUDirect-TCPX"
+  if [[ "$USE_GPUDIRECT" == "tcpx" ]] || [[ "$USE_GPUDIRECT" == "fastrak" ]]; then
+    export CUDA_DEVICE_MAX_CONNECTIONS=1
+    export NCCL_ALGO=Ring,Tree
```
[the remark this answers is not in the document]
Comment (michelle-yooh) [reply gpu_multi_process_run.sh]: It worked on A3 but I only tested a simple run. It makes sense to me to add it later once it's fully verified.

File: gpu_multi_process_run.sh (hunk 2 above)
Comment (michelle-yooh) [reply gpu_multi_process_run.sh]: Right. Removed for a3+

**Review by yangyuwei — APPROVED** [review]

**Pull request merged**